src/workspaceState.ts

```diff
@@ -0,0 +1,73 @@
+import * as vscode from 'vscode'
+
+type WorkspaceStateKey = 'LastActiveManifestUri' | 'Initialized' | 'DependenciesUpdated' | 'DependenciesBuilt' | 'ApplicationBuilt'
+
+export class WorkspaceState {
```

Comment:
Would be nice to have small comments to explain what is this used for, just to make it easier for potential future contributors